Source/Win32/arcana/threading/task_schedulers.h

```diff
@@ -0,0 +1,40 @@
+//
+// Copyright (C) Microsoft Corporation. All rights reserved.
+//
+
+#pragma once
+
+#include <Windows.h>
+#include <functional>
+#include <arcana/win32_exception.h>
+
+namespace arcana
+{
+    // NOTE: These task schedulers are for the arcana task system.
+
+    namespace
+    {
+        constexpr struct
+        {
+            template<typename CallableT>
+            void operator()(CallableT&& callable) const
+            {
+                auto callback = [](PTP_CALLBACK_INSTANCE instance, PVOID context, PTP_WORK work)
+                {
+                    auto callable_ptr = std::unique_ptr<CallableT>(static_cast<CallableT*>(context));
+                    (*callable_ptr)();
+                    CloseThreadpoolWork(work);
+                };
+
+                auto callable_ptr = std::make_unique<CallableT>(callable);
```

Comment:
This allocates on the heap. I can't think of a way to avoid this unless we do some kind of callback pooling. Anyone have ideas?